services/QuillLMS/app/models/app_setting.rb

```diff
@@ -0,0 +1,59 @@
+# == Schema Information
+#
+# Table name: app_settings
+#
+#  id                 :integer          not null, primary key
+#  enabled            :boolean          default(FALSE), not null
+#  enabled_for_admins :boolean          default(FALSE), not null
+#  name               :string           not null
+#  pct_active         :integer          default(0), not null
+#  user_whitelist     :string           default([]), not null, is an Array
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+# Indexes
+#
+#  index_app_settings_on_name  (name)
+#
+class AppSetting < ActiveRecord::Base
+    PCT_ACTIVE_VALUES = [100, 90, 80, 70, 60, 50, 40, 30, 20, 10] 
+
+    validates :pct_active, inclusion: { in: PCT_ACTIVE_VALUES}
+    validates :name, uniqueness: true
+
+    def self.enabled?(name:, user:)
+        as = AppSetting.find_by_name name
+        return false unless as 
+        return false unless user.respond_to? :id 
+        as._enabled? user.id
+    end
+
+    def _enabled? user
```

Comment:
Normal development should only use `enabled?`, because it's safe to invoke with bad input (non existent user or AppSetting records). The reason why I'm not making it a private method is because the controller needs to invoke `_enabled?`